packages/styling/src/utilities/icons.ts

```diff
@@ -108,11 +119,47 @@ export function registerIcons(iconSubset: IIconSubset, options?: Partial<IIconOp
   }
 }
 
+/**
+ * Unregisters icons by name.
+ *
+ * @param iconNames - List of icons to unregister.
+ */
+export function unregisterIcons(iconNames: string[]): void {
+  const options = _iconSettings.__options;
+
+  for (const iconName of iconNames) {
+    const normalizedIconName = normalizeIconName(iconName);
+    if (_iconSettings[normalizedIconName]) {
+      delete _iconSettings[normalizedIconName];
+    } else {
+      // Warn that we are trying to delete an icon that doesn't exist
+      if (!options.disableWarnings) {
+        warn(`The icon "${iconName}" tried to unregister but was not registered.`);
+      }
+    }
+
+    // Delete any aliases for this iconName
+    if (_iconSettings.__remapped[normalizedIconName]) {
+      delete _iconSettings.__remapped[normalizedIconName];
+    }
+
+    // Delete any items that were an alias for this iconName
+    const remappedKeys = Object.keys(_iconSettings.__remapped).filter((key: string) => {
+      return _iconSettings.__remapped[key] === normalizedIconName;
+    });
+    if (remappedKeys.length) {
+      for (const remappedKey of remappedKeys) {
```

Comment:
Maybe we should just do this in a single pass instead of filter then delete?

Reply:
Yeah I wasn't sure if it would cause an error by modifying the dictionary while filtering, but I guess it's just filtering through the created array of keys so there shouldn't be an issue